owasp-top10-2021-apps/a9/games-irados/app/routes.py

```diff
@@ -76,10 +119,20 @@ def login():
         username = request.form.get('username').encode('utf-8')
         psw = Password(request.form.get('password').encode('utf-8'))
         user_password, success = database.get_user_password(username)
-        if not success or user_password == None or not psw.validate_password(str(user_password[0])):
+        if not success:
+            if isinstance(user_password, str):
+                flash("Database error", "danger")
+                logfile.info("Unsuccessful login attempt (database error) - Username: {} - Database message: {} - IP: {}".format(username, user_password, request.remote_addr))
```

Comment:
get_user_password() sends password and Mysql data using the same variable. The difference is in the type.
str for message and tuple for password.